### ISSUE TITLE
In flowcharts, how to get the graph lines to come out of vertices of the decision box

### ISSUE THREAD

**Issue by rishavs** [mermaid-js/mermaid#3260]
Using the example in mermaid.live for flowcharts, I can create a nice flowchart - but the decision box ("is true?") has the graph arrows to the next steps coming out of its body.
Here is what I get today
![image](https://user-images.githubusercontent.com/1210239/180772401-e529eca9-cf7f-403e-9204-170e191a2fa9.png)

Instead, I would want the lines to come out of the vertices of the decision box - something like this;
![image](https://user-images.githubusercontent.com/1210239/180772787-294d1c85-8fc5-4f3a-b958-fdb8ccb95e3a.png)

Is there a way to define my charts to push the graphing lines from the vertices today? If not, can I make this Feature Request?
The flowcharts just feel far more organized and pleasing that way.

**Comment by flywire** [comment]
Maybe by setting `curve`, see https://stackoverflow.com/questions/72538993/mermaid-flowchart-line-style.

Include your code so others can test.

**Comment by rishavs** [comment]
Hi @flywire . The code I used is the sample chart code in mermaid live editor for flowcharts.

```mermaid
graph TD
    A[Christmas] -->|Get money| B(Go shopping)
    B --> C{Let me think}
    C -->|One| D[Laptop]
    C -->|Two| E[iPhone]
    C -->|Three| F[fa:fa-car Car]
```

**Comment by flywire** [comment]
No, but `%%{init: {'flowchart' : {'curve' : 'stepAfter'}}}%%` is pretty close:

[![image](https://user-images.githubusercontent.com/11288701/183363687-391855d4-bf0c-444f-a1ec-e3d964d6d0f4.png)](https://mermaid.live/edit#pako:eNpVkM1qw0AMhF9FLBi3EL-AD4XE-bkEWmhuXh-ELWeXen9Yyw3B3nfvummh1WnEfDMgzaJ1HYlSZNmsreYS5rwf3K1VGDiHdW2n8EmrzEcmv-2ZQh5jzDJppb0G9Aoue2khzbauVNAjGxwbKIqX5UQMxlm6L7B7OjkYlfNe2-vzg9-tEFTzecUIWGn7ER9W9Z1_tbTAvj6jZ-ebv87l5hY41PpNpfr_jgqUUse6x7LHosUAFYZGbIShYFB36dp5DUjBigxJUSbZUY_TwFJIGxM6-Q6ZDp1mF0SqGUbaCJzYvd9tK0oOE_1Ce43pC-aHil8LqW5s)
Click image to open in mermaid.live


Using `%%{init: {'flowchart' : {'curve' : 'stepBefore'}}}%%`:

[![image](https://user-images.githubusercontent.com/11288701/183365867-4f865a4e-451f-4373-a508-aff3a4ea2a95.png)](https://mermaid.live/edit#pako:eNpNzk0KgzAQhuGrhAGJgi7aZXZaPUG7zCaYsQY0kThpKZK7V_tnd8_HvItZoHUaQUCSLMYaEmzh3eDuba88cbbNNvgbbuQz4VRh5zzyGGOSSCvtL2bsUkvLWJmmZZa9xIqiYNXO0856Z7OxereHPx8hhxH9qIxe_1u2iwTqcUQJYqXGToWBJEgb1zRMWhE22pDzIDo1zJiDCuTOD9uCIB_wG9VGXb0aP1V8AhJWTrI)
Click image to open in mermaid.live

The bug seems to be the lines don't originate from the centre of the node.

In a normal flowchart there should be a single path from the node before they diverge. A decision node is different.

**Comment by rishavs** [comment]
> The bug seems to be the lines don't originate from the centre of the node.

That might be it. I think just having that as a config might be good too.

**Comment by flywire** [comment]
[Linear Curve](https://stackoverflow.com/a/73757144/4539999) is better:

![image](https://user-images.githubusercontent.com/11288701/196830696-6f969b91-61da-42f2-80d9-b2e64f06a7f4.png)

**Comment by sidharthv96** [comment]
`flowchart-elk` will do the job. 

<img width="1512" alt="image" src="https://github.com/mermaid-js/mermaid/assets/10703445/897daf13-616a-4c1a-aff4-ea6bf2a5e4d0">

```mermaid
flowchart-elk TD
    A[Christmas] -->|Get money| B(Go shopping)
    B --> C{Let me think}
    C -->|One| D[Laptop]
    C -->|Two| E[iPhone]
    C -->|Three| F[fa:fa-car Car]
```

**Comment by flywire** [comment]
> `flowchart-elk` will do the job.

No, it badly screws the order - click image for code.

**Original:**

[
![image](https://github.com/mermaid-js/mermaid/assets/11288701/9b3c9095-e586-4711-a149-48934db9178b)
](https://mermaid.live/edit#pako:eNpNzbEOwiAUheFXIXdqkzLoyNaKT6DjXW7g1jYWMAgxpum7S3XA7UvOn5wVTLAMCsYlvMxEMYmrRi9E3zR9234lpJRiqDxV6srzzuHXHv58RA8dOI6OZluO1n1DSBM7RlCFluIdAf1WOsopXN7egEoxcwf5YSmxnukWyYEaaXny9gEKXjOA)

**-elk:**

[![image](https://github.com/mermaid-js/mermaid/assets/11288701/f9774dea-8c30-4129-a623-7281b992d232)](https://mermaid.live/edit#pako:eNpNzb0KwjAUhuFbCWdqoRl0zNYar0DHLIfk1JbmR9IEkdJ7N9Uhbg98L3wb6GAIBIw2vPSEMXGyC7tL5Rnrm6Zv268Y55wNlZdKWXk9OPza05_PykMHjqLD2ZSz7dgUpIkcKRCFBuOiQPm9dJhTuL29BpFipg7y02AiOeMjogMxol1p_wANcjTp)

**Comment by sidharthv96** [comment]
<img width="375" alt="image" src="https://github.com/mermaid-js/mermaid/assets/10703445/f5a2afb5-1049-4622-b000-818755fed52c">

With even number of nodes. I think tweaking the elk rendering logic should fix this.
